src/bootstrap/test.rs

```diff
@@ -1284,6 +1293,63 @@ macro_rules! test_definitions {
     };
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, Ord, PartialOrd)]
+pub struct RunMakeSupport {
+    pub compiler: Compiler,
+    pub target: TargetSelection,
+}
+
+impl Step for RunMakeSupport {
+    type Output = PathBuf;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.path("src/tools/run-make-support")
+    }
+
+    fn make_run(_run: RunConfig<'_>) {}
```

Comment:
I'm not 100% this is correct, as I had to move the ensure stage1-std dep to where run-make was getting built to make this work again.

Reply:
`should_run` without `make_run` is kind of useless. `should_run` determines when `make_run` should be called.

Reply:
I'm not sure what I should change here, since I had to move the following

```rs
if $mode == "run-make" {
    let stage1_compiler = run.builder.compiler(1, run.builder.config.build); // this
    run.builder.ensure(compile::Std::new(stage1_compiler, run.target)); // and this
    run.builder.ensure(RunMakeSupport {
        compiler: stage1_compiler,
        target: run.build_triple(),
    });
}
```

to the `make_run` of `RunMake` generated by `test_definitions!` in order to actually get stage1-std to build before `run-make-support` is being built. For some reason, trying to ensure stage1-std is being built within `RunMakeSupport`'s `make_run` doesn't seem to work?

Reply:
I'm not sure what the issue is here. Maybe ask on zulip?

Reply:
I think the current way to ensure stage1-std is built before `run_make_support` is probably  more correct.

Reply:
In that case could you remove the `should_run` condition?

Reply:
> In that case could you remove the `should_run` condition?

Sorry for the late reply, do you mean changing it to be something like `run.never()` inside `should_run`?

Reply:
Yes